chain/stmgr/forks.go

```diff
@@ -208,7 +211,21 @@ func (sm *StateManager) handleStateForks(ctx context.Context, root cid.Cid, heig
 	return retCid, nil
 }
 
-func (sm *StateManager) hasExpensiveFork(ctx context.Context, height abi.ChainEpoch) bool {
+// Returns true if executing the current tipset would trigger an expensive fork.
+//
+// - If the tipset is the genesis, this function always returns false.
```

Comment:
As written now this invariant depends on the sm.expensiveUpgrades table which in reasonable scenarios could potentially contain "expensive" upgrades if say we did all migrations at the genesis epoch. 

It might be better to override the table to always return false on 0 to maintain this invariant.  Any network doing this would probably be doing very little work in the expensive migration since nothing has happened in state yet.

Also all existing callers seem to implicitly assume that this is the case anyway by only checking heights > 0 so why not make it official.

Reply:
Because the function is called `hasExpensiveFork`, not `hasExpensiveForkOrIsGenesis` :smile:. Basically, I don't want to surprise users.

At the moment, genesis matters to all callers because there's really no other option. But I'd like to keep that as an explicit decision.

Reply:
But I've fixed the issue in Call where I wasn't handling this consistently.